test/models/abstract_model_test.rb

```diff
@@ -395,4 +395,37 @@ def test_species_list_rss_log_life_cycle
     assert_nil(SpeciesList.safe_find(spl_id))
     assert_equal(:species_list, rss_log.target_type)
   end
+
+  # -------------------------------------------------------------------
+  #  Test the methods that generate controller and action names
+  #  based on the Query model
+  # -------------------------------------------------------------------
+
+  def test_show_controller
+    assert_equal("articles", Article.show_controller)
+    assert_equal("#{self.class.name.underscore}/phony", Phony.show_controller)
```

Comment:
Is there a better way to test for a controller with a singular name?
I didn't want to use any of the current actual controllers because they may be normalized, which would break the test.